critiquebrainz/frontend/views/mapping.py

```diff
@@ -175,16 +181,22 @@ def parse_spotify_id(spotify_ref):
     Supported reference types:
       - Spotify URI (spotify:album:6IH6co1QUS7uXoyPDv0rIr)
       - HTTP link (http://open.spotify.com/album/6IH6co1QUS7uXoyPDv0rIr)
+
+    Returns:
+        parsed spotify id (ex. "6IH6co1QUS7uXoyPDv0rIr") if supported reference types are provided, else None
     """
-    # Spotify URI
-    if spotify_ref.startswith('spotify:album:'):
-        return spotify_ref[14:]
-
-    # Link to Spotify
-    # TODO(roman): Improve checking there.
-    if spotify_ref.startswith('http://') or spotify_ref.startswith('https://'):
-        if spotify_ref.endswith('/'):
-            spotify_ref = spotify_ref[:-1]
-        return os.path.split(urllib.parse.urlparse(spotify_ref).path)[-1]
-
-    # TODO(roman): Raise exception if failed to parse!
+    try:
```

Comment:
We should raise exceptions if the spotify reference is invalid also. And these exceptions should be handled in the view which calls this function.